Zip/Functions/Add-ZipArchiveEntry.ps1

```diff
@@ -159,6 +153,53 @@ function Add-ZipArchiveEntry
                 $stream.Dispose()
             }
         }
+
+        $filePaths = $InputObject | Resolve-Path | Select-Object -ExpandProperty 'ProviderPath'
+        foreach( $filePath in $filePaths )
+        {
+            Write-Debug -Message $filePath
+
+            if( $BasePath )
+            {
+                $baseEntryName = $filePath -replace $basePathRegex,''

```

Comment:
Something I just noticed, when we construct the `$basePathRegex` on [line 101](https://github.com/webmd-health-services/Zip/blob/8ab06e4b060144c9c083a9c4d471c742518bcab2/Zip/Functions/Add-ZipArchiveEntry.ps1#L101), we append a `[IO.Path]::DirectorySeparatorChar`. Do we need to first trim the given value, e.g. `$BasePath.Trim([IO.Path]::DirectorySeparatorChar)`? What if someone gave `C:\projects\zip\` as the `BasePath`, would the pattern become `^C:\\projects\\zip\\\\?` (extra escaped `\` char)? Or is this handled somewhere already?